pkg/test/framework/components/namespace/kube.go

```diff
@@ -117,3 +109,21 @@ func newKube(ctx resource.Context, prefix string, inject bool) (Instance, error)
 
 	return n, nil
 }
+
+// createNamespaceLabels will take a namespace config and generate the proper k8s labels
+func createNamespaceLabels(cfg *Config) map[string]string {
+	l := make(map[string]string)
+
+	// bring over existing component labels
+	for k, v := range cfg.ComponentLabels {
+		l[k] = v
+	}
+
+	if cfg.Inject {
```

Comment:
I think you should do this first, then apply the generic labels. Reason being that a user could theoretically not set `cfg.Inject` in the config, but set the `istio-injection` label.

Reply:
fixed